system-addon/lib/TopSitesFeed.jsm

```diff
@@ -131,7 +137,12 @@ this.TopSitesFeed = class TopSitesFeed {
     // Now, get a tippy top icon, a rich icon, or screenshot for every item
     for (const link of withPinned) {
       if (link) {
-        this._fetchIcon(link);
+        // If there is a custom screenshot this is the only image we display
+        if (link.customScreenshotURL && !link.customScreenshot) {
+          this._fetchCustomScreenshot(link);
+        } else {
```

Comment:
Does the previous condition actually want nested `if` blocks? In particular, if we have a custom screenshot url and already have the screenshot, we would end up fetching the icon?

Reply:
Alternatively, the existence check should happen inside `_fetch` similar to `_fetchIcon`

Reply:
Kept the check for customScreenshotURL,`_fetchIcon` is not needed if a custom url is present.